TACC-Cloud/geoapi
# task/WG-44: Updating PyJWT package

## Overview: ##
Updating PyJWT package version from 1.7.1 to ^2.0.0 (in our case it updated to the latest version 2.8.0)
## Related Jira tickets: ##

* [WG-44](https://tacc-main.atlassian.net/browse/WG-44)

## Summary of Changes: ##

- In the pyproject.toml file I changed the pyJWT version from 1.7.1 to ^2.0.0 (which will use any version of 2.0.0 and up but not 3.0.0)
- I ran poetry update pyjwt command inside the geoapiworkers container so that the poetry.lock file would install the new version and any necessary dependencies/package extras for pyJWT. 
- Poetry also added additional code for all packages where it list the package file location and hash location
- In order for the JWT verification to work with version 2.8.0, I had to update the codebase to use the new parameter (verify was depreciated). It's now the options param with a key-value pair of verify_signature as a boolean. 
- Because we don't want to verify in the testing env, I made a conditional in the codebase to not verify when the env is TESTING

## Testing Steps: ##
1. Run 'make build' so poetry installs the new versions it needs for these changes.
2. You'll need your real JWT (ask Nathan or I if you hadn't obtained this yet) in order for verification to work with this code and new pyJWT version locally.
3. Make sure you can see your Project Listings in local env (means verification worked correctly)
4. Run all three unit tests (geoapi, geoapi workers and hazmapper) and make sure all tests pass.
5. Look for any errors/compatibility issues in general that might be caused by the new packages/versions.

## Notes: ##
